pkg/collect/result.go

```diff
@@ -18,13 +18,72 @@ func NewResult() CollectorResult {
 	return map[string][]byte{}
 }
 
+// SymLinkResult creates a symlink (relativeLinkPath) of relativeFilePath in the bundle. If bundlePath
+// is empty, no symlink is created. The relativeLinkPath is always saved in the result map.
+func (r CollectorResult) SymLinkResult(bundlePath, relativeLinkPath, relativeFilePath string) error {
+	// We should have saved the result this symlink is pointing to prior to creating it
+	klog.Info("Creating symlink ", relativeLinkPath, " -> ", relativeFilePath)
+	data, ok := r[relativeFilePath]
+	if !ok {
+		return errors.Errorf("cannot create symlink, result in %q not found", relativeFilePath)
+	}
+
+	if bundlePath == "" {
+		// If we are not saving results to disk, cache in memory
+		r[relativeLinkPath] = data
+		return nil
+	}
+
+	linkPath := filepath.Join(bundlePath, relativeLinkPath)
+	filePath := filepath.Join(bundlePath, relativeFilePath)
+
+	// If both paths are the same, don't create a symlink
+	if linkPath == filePath {
+		return nil
+	}
+
+	linkDirPath := filepath.Dir(linkPath)
+
+	// Create the directory for the symlink if it doesn't exist
+	err := os.MkdirAll(linkDirPath, 0777)
+	if err != nil {
+		return errors.Wrap(err, "failed to create directory")
+	}
+
+	// Ensure the file exists
+	_, err = os.Stat(filePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to stat. File may not exist")
+	}
+
+	// Do nothing if the symlink already exists
+	_, err = os.Lstat(linkPath)
+	if err == nil {
+		return nil
+	}
+
+	// Create the symlink
+	err = os.Symlink(filePath, linkPath)
```

Comment:
Question: does this create a link that has the entire directory path when it's tar'd or does it have the appropriate `../../foo/bar` type of relative path?

Reply:
> 

It does not. That is handled when creating the tar archive [here](https://github.com/banjoh/troubleshoot/blob/em/sbctl-pod-logs-744/pkg/collect/result.go#L256). We also do not want to do that here cause the view of the structure needs to remain intact for use cases such as running preflights where the collected data is only used in-memory and never archived.

Reply:
I added a code comment